Single owner can manage delegates to mint

edgeoftheworld opened the pull request:
removed voting functions, single owner who manages minters, owner can change faucet contract on token

edgeoftheworld committed:
Single owner can manage delegates to mint

removed voting functions, single owner who manages minters, owner can change faucet contract on token

edgeoftheworld commented:
here is the etherscan link to the deployed contract https://rinkeby.etherscan.io/address/0x29686429aad2b0674ed36a4d95b66c299dafc645

fubuloubu commented:
Some update suggestions:
1. Make the delegate data structure into a map e.g. `mapping (address => bool) public canMint`
NOTE `public` creates a getter:
```
// Automatically generated by solidity:
function canMint(address _delegate) public view returns (bool) {
    return canMint[_delegate];
}
```
2. `owner` can control delegate's minting abilities (through `canMint`)
3. Add an event to track changes to access rights (for UI) e.g:
`event AccessChanged(address delegate, bool canMint)`
4. Set `canMint` of `owner` to true in the constructor

Commit:
Changed delegates to mapping, added event for AccessControl

changed delegates to mapping from array to avoid looping.  Added event when access control is changed.

Commit:
Removed uncalled function

isDelegate function no longer used with mapping

edgeoftheworld commented:
updated contract on Rinkeby.  https://rinkeby.etherscan.io/address/0x5f1b069d88d239a52f51a6a7c5ecdfeec0afcbff

fubuloubu commented on contracts/DappDevMint.sol:
```diff
@@ -88,94 +58,40 @@ contract DappDevMint
+      //only owner can add delegate
+      require(isOwner(msg.sender));
+      canMint[newDelegate] = true;
+      emit AccessControl(newDelegate, true);
```
This is a new feature for v0.4.21, make sure the pragma reflects this

fubuloubu replied:
“‘emit’ keyword in Solidity” @AniketEngg https://medium.com/@aniketengg/emit-keyword-in-solidity-242a679b0e1a

fubuloubu commented:
Looking good! Will have to test next week!

Commit:
Updated compiler version

updated compiler to 0.4.21 for support of emit keyword for events

edgeoftheworld commented:
compiler version changed.
https://rinkeby.etherscan.io/address/0x616f47efc6dcd6ca09c08426c7111c51a5dd2d10
